test/rekt/features/pingsource/features.go

```diff
@@ -129,3 +130,27 @@ func SendsEventsWithEventTypes() *feature.Feature {
 
 	return f
 }
+
+func SendsOneOffsEvent() *feature.Feature {
+	source := feature.MakeRandomK8sName("oneoff pingsource")
+	sink := feature.MakeRandomK8sName("sink")
+	f := feature.NewFeature()
+
+	f.Setup("install sink", eventshub.Install(sink, eventshub.StartReceiver))
+
+	cfg := []manifest.CfgFn{
+		pingsource.WithDate((time.Now().Add(1 * time.Minute)).Format("2006-01-02T15:04:05.000Z")),
+		pingsource.WithDataBase64("text/plain", "aGVsbG8sIHdvcmxkIQ=="),
+		pingsource.WithSink(svc.AsKReference(sink), ""),
+	}
+
+	f.Requirement("install pingsource", pingsource.Install(source, cfg...))
+	f.Requirement("pingsource goes ready", pingsource.IsReady(source))
+
+	f.Stable("OneOff PingSource as event source").
+		Must("delivers events", assert.OnStore(sink).MatchEvent(
+			test.HasType("dev.knative.sources.ping"),
+		).Exact(1))
```

Comment:
Can we assert that, only after we have received the event successfully, the PingSource is completed?

Something like:
```golang
Must("delivers events and be completed", assertEventReceivedAndPingSourceComplted(source, sink, test.HasType("dev.knative.sources.ping"))

// ...

func assertEventReceivedAndCompleted(source, sink string, test.EventMatcher) feature.StepFn {
	return func(ctx context.Context, t feature.T) {
		assert.OnStore(sink).MatchEvent(test.HasType("dev.knative.sources.ping")).Exact(1)(ctx, t)
		// pseudo code
		// get ping source `source`
		// verify it is marked as completed
	}
}
```

Reply:
Got it